plugin/pkg/shoot/binding/admission.go

```diff
@@ -0,0 +1,268 @@
+type Binding struct {
+	*admission.Handler
+	seedLister       corelisters.SeedLister
+	shootLister      corelisters.ShootLister
+	shootStateLister corev1alpha1listers.ShootStateLister
+	readyFunc        admission.ReadyFunc
+}
+
+var (
+	_ = admissioninitializer.WantsInternalCoreInformerFactory(&Binding{})
+	_ = admissioninitializer.WantsExternalCoreInformerFactory(&Binding{})
+
+	readyFuncs = []admission.ReadyFunc{}
+)
+
+// New creates a new Binding admission plugin.
+func New() (*Binding, error) {
+	return &Binding{
+		Handler: admission.NewHandler(admission.Create),
+	}, nil
+}
+
+// AssignReadyFunc assigns the ready function to the admission handler.
+func (b *Binding) AssignReadyFunc(f admission.ReadyFunc) {
+	b.readyFunc = f
+	b.SetReadyFunc(f)
+}
+
+// SetInternalCoreInformerFactory gets Lister from SharedInformerFactory.
+func (b *Binding) SetInternalCoreInformerFactory(f coreinformers.SharedInformerFactory) {
+	seedInformer := f.Core().InternalVersion().Seeds()
+	b.seedLister = seedInformer.Lister()
+
+	shootInformer := f.Core().InternalVersion().Shoots()
+	b.shootLister = shootInformer.Lister()
+
+	readyFuncs = append(
+		readyFuncs,
+		seedInformer.Informer().HasSynced,
+		shootInformer.Informer().HasSynced,
+	)
+}
+
+// SetExternalCoreInformerFactory sets the external garden core informer factory.
+func (b *Binding) SetExternalCoreInformerFactory(f externalcoreinformers.SharedInformerFactory) {
+	shootStateInformer := f.Core().V1alpha1().ShootStates()
+	b.shootStateLister = shootStateInformer.Lister()
+
+	readyFuncs = append(readyFuncs, shootStateInformer.Informer().HasSynced)
+}
+
+// ValidateInitialization checks whether the plugin was correctly initialized.
+func (b *Binding) ValidateInitialization() error {
+	if b.seedLister == nil {
+		return errors.New("missing seed lister")
+	}
+	if b.shootLister == nil {
+		return errors.New("missing shoot lister")
+	}
+	if b.shootStateLister == nil {
+		return errors.New("missing shoot state lister")
+	}
+	return nil
+}
+
+var _ admission.ValidationInterface = &Binding{}
+
+// Validate validates the Binding against the existing conditions.
+func (b *Binding) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	// Wait until the caches have been synced
+	if b.readyFunc == nil {
+		b.AssignReadyFunc(func() bool {
+			for _, readyFunc := range readyFuncs {
+				if !readyFunc() {
+					return false
+				}
+			}
+			return true
+		})
+	}
+	if !b.WaitForReady() {
+		return admission.NewForbidden(a, errors.New("not yet ready to handle request"))
+	}
+
+	// Ignore all kinds other than Binding
+	if a.GetKind().GroupKind() != core.Kind("Binding") {
+		return nil
+	}
+
+	binding, convertIsSuccessful := a.GetObject().(*core.Binding)
+	if !convertIsSuccessful {
+		return apierrors.NewInternalError(errors.New("could not convert resource into Shoot object"))
+	}
+
+	if len(binding.Target.Kind) != 0 && binding.Target.Kind != "Seed" {
```

Comment:
This should probably be:
```suggestion
	if len(binding.Target.Kind) == 0 || binding.Target.Kind != "Seed" {
```
So that a `Target` with empty `Kind` also reports an error?